src/orderbook/state.ts

```diff
@@ -28,7 +83,139 @@ export class AccountState {
       )
     }
     this.batchId = batchId
-    // TODO(nlordell): Actually apply events to state.
+
+    for (const ev of events) {
+      const data = ev.returnValues
+      switch (ev.event) {
+        case 'OrderCancellation':
+          this.order(data.owner, parseInt(data.id)).validUntil = (await this.batches.forEvent(ev)) - 1
+          break
+        case 'OrderDeletion':
+          this.deleteOrder(data.owner, parseInt(data.id))
+          break
+        case 'OrderPlacement':
+          this.addOrder(data.owner, {
+            buyToken: parseInt(data.buyToken),
+            sellToken: parseInt(data.sellToken),
+            validFrom: parseInt(data.validFrom),
+            validUntil: parseInt(data.validUntil),
+            priceNumerator: BigInt(data.priceNumerator),
+            priceDenominator: BigInt(data.priceDenominator),
+          })
+          break
+        case 'TokenListing':
+          this.addToken(parseInt(data.id), data.token)
+          break
+        case 'Deposit':
+        case 'SolutionSubmission':
+        case 'Trade':
+        case 'TradeReversion':
+        case 'WithdrawRequest':
+        case 'Withdraw':
+          // TODO(nlordell): Handle more events.
+          break
```

Comment:
Am I to read that, so far, only the `TokenListing` has been handled thus far?

Reply:
Yes, I split up this PR into 2. Otherwise it would have been WAY to big.